apps/dashboard/app/lib/smart_attributes/attributes/auto_output_directory.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+  class AttributeFactory
+    # Build this attribute object. Must specify a valid directory in opts
+    #
+    # @param opts [Hash] attribute's options
+    # @return [Attributes::AutoOutputDirectory] the attribute object
+    def self.build_auto_output_directory(opts = {})
+      Attributes::AutoOutputDirectory.new('auto_output_directory', opts)
+    end
+  end
+
+  module Attributes
+    class AutoOutputDirectory < Attribute
+      # Value of auto_output_directory attribute
+      # Defaults to first script path in the project
+      # @return [String] attribute value
+      def value
+        opts[:value] || 'Default Output Directory'
+      end
+
+      def widget
+        'text_field'
+      end
+
+      def label(*)
+        (opts[:label] || 'Output Directory').to_s
+      end
+
+      def output_directory_hash
+        { output_path: output_path_value, error_path: error_path_value  }
+      end
+
+      # Submission hash describing how to submit this attribute
+      # @param fmt [String, nil] formatting of hash
+      # @return [Hash] submission hash
+      def submit(*)
+        opts[:value] ? { script: output_directory_hash } : nil
```

Comment:
This won't work. What we want to return is a hash with the correct key and value pairs. That hash is then used to create a `Script` object in ood_core below.

That hash needs to have these 2 keys so that the resulting Script object has those attributes. So the result should actually look something like this:

```ruby
{ script: { output_path: value, error_path: value } }
```

https://github.com/OSC/ood_core/blob/4342e29119c9db3a4860940f83191ba1cb760156/lib/ood_core/job/script.rb#L67

https://github.com/OSC/ood_core/blob/4342e29119c9db3a4860940f83191ba1cb760156/lib/ood_core/job/script.rb#L71


https://github.com/OSC/ood_core/blob/4342e29119c9db3a4860940f83191ba1cb760156/lib/ood_core/job/script.rb